bundles/org.openhab.binding.sagercaster/src/main/java/org/openhab/binding/sagercaster/internal/handler/SagerCasterHandler.java

```diff
@@ -0,0 +1,267 @@
+                    } else {
+                        logger.debug("Channel '{}' only accepts OnOffType commands.", channelUID);
+                    }
+                    break;
+                case CHANNEL_WIND_SPEED:
+                    logger.debug("Updated wind speed, updating forecast");
+                    if (command instanceof DecimalType) {
+                        DecimalType newValue = (DecimalType) command;
+                        scheduler.submit(() -> {
+                            sagerWeatherCaster.setBeaufort(newValue.intValue());
+                            postNewForecast();
+                        });
+                    } else {
+                        logger.debug("Channel '{}' only accepts DecimalType commands.", channelUID);
+                    }
+                    break;
+                case CHANNEL_PRESSURE:
+                    logger.debug("Sea-level pressure updated, updating forecast");
+                    if (command instanceof QuantityType) {
+                        @SuppressWarnings("unchecked")
+                        QuantityType<Pressure> newPressure = ((QuantityType<Pressure>) command)
+                                .toUnit(HECTO(SIUnits.PASCAL));
+                        if (newPressure != null) {
+                            pressureCache.put(newPressure);
+                            QuantityType<Pressure> agedPressure = pressureCache.getAgedValue();
+                            if (agedPressure != null) {
+                                scheduler.submit(() -> {
+                                    sagerWeatherCaster.setPressure(newPressure.doubleValue(),
+                                            agedPressure.doubleValue());
+                                    updateChannelString(GROUP_OUTPUT, CHANNEL_PRESSURETREND,
+                                            String.valueOf(sagerWeatherCaster.getPressureEvolution()));
+                                    postNewForecast();
+                                });
+                            } else {
+                                updateChannelString(GROUP_OUTPUT, CHANNEL_FORECAST, FORECAST_PENDING);
+                            }
+                        }
+                    }
+                    break;
+                case CHANNEL_TEMPERATURE:
+                    logger.debug("Temperature updated");
+                    if (command instanceof QuantityType) {
+                        @SuppressWarnings("unchecked")
+                        QuantityType<Temperature> newTemperature = ((QuantityType<Temperature>) command)
+                                .toUnit(SIUnits.CELSIUS);
+                        if (newTemperature != null) {
+                            temperatureCache.put(newTemperature);
+                            currentTemp = newTemperature.intValue();
+                            QuantityType<Temperature> agedTemperature = temperatureCache.getAgedValue();
+                            if (agedTemperature != null) {
```

Comment:
Just for my understanding: This part inside the `if` differs a little bit from other implementations (e.g. pressure -> usage of `scheduler` and `postNewForecast()`) Is that correct?

Reply:
Yes, because trend interpretation for pressure is done based on the SagerCaster weather forecast algorithm for its internal needs, while trend observation for temperture is just an added feature I delivered because I could make it available easily.